add support for Raspberry Pi Pico

K4KDR opened the pull request:
added support for Raspberry Pi Pico microcontroller in 'contributed' folder.

** ref. the command syntax that is used w/ this board:

`SX1262 radio = new Module(3, 20, 15, 2, SPI1, RADIOLIB_DEFAULT_SPI_SETTINGS);`

UPDATE on 3-June-2024:    removed my request here for help with custom SPI... I remembered that the correct syntax for this board had already been determined from https://github.com/jgromes/RadioLib/issues/729      

However, while the provide syntax works in the sketch itself, I am not sure if I have positioned it properly in this Pull Request.   So, a close review is still required & appreciated!

Thank you!

K4KDR committed:
Create README.md

Primarily to also create the 'contributed' folder

K4KDR committed:
Create PI_PICO.h

add support for Raspberry Pi Pico microcontroller

K4KDR committed:
Update RadioBoards.h

added support for 'RADIO_BOARD_RASPBERRYPI_PICO'

K4KDR committed:
Update PI_PICO.h

removed temp line from earlier edit

K4KDR committed:
Update PI_PICO.h

updated to include confirmed good SPI settings & commands for this board

jgromes commented on src/contributed/RaspberryPi/PI_PICO.h:
```diff
@@ -0,0 +1,29 @@
+#define RADIO_SCK     (10)
+
+#define RADIO_SPI_INIT                                    \
+SPI1.setSCK(10);         \       // from https://github.com/jgromes/RadioLib/issues/729
+SPI1.setTX(11);          \       // ~
+SPI1.setRX(12);          \       // ~  
```
I think it would be better to use the defined macros here, e.g. `RADIO_SPI.setSCK(RADIO_SCK);` instead

jgromes commented on src/contributed/RaspberryPi/PI_PICO.h:
```diff
@@ -0,0 +1,29 @@
+SPI1.setSCK(10);         \       // from https://github.com/jgromes/RadioLib/issues/729
+SPI1.setTX(11);          \       // ~
+SPI1.setRX(12);          \       // ~  
+pinMode(3, OUTPUT);      \       // ~
+digitalWrite(3, HIGH);   \       // ~
```
This is redundant, RadioLib initializes the NSS.

jgromes commented on src/contributed/RaspberryPi/PI_PICO.h:
```diff
@@ -0,0 +1,29 @@
+SPI1.setRX(12);          \       // ~  
+pinMode(3, OUTPUT);      \       // ~
+digitalWrite(3, HIGH);   \       // ~
+RADIO_SPI1.begin(false);         // from https://github.com/jgromes/RadioLib/issues/729
```
This should probably be `RADIO_SPI.begin(false);`, I'm guessing a typo?

jgromes reviewed (changes requested):
Thank you! I left some comments.

Also, could you please add a section for the wireless module type? I added that in the latest commit.

```
#if RADIOLIB_SUPPORT_ENABLED
  #define Radio       SX1262
#endif
```

K4KDR committed:
Update PI_PICO.h

attempted to make requested changes.    HUGE thanks for your patience as I am only familiar w/ a small percentage of what is proper syntax.   However, with each post you all share a tremendous amount of education to anyone who sees these updates, so thank you!

jgromes commented:
All looks good now - merged, thank you!

serialrf433 commented:
This merged PR opens up the question if just any generic SBC that have no RadioLib-RF by default would get merged here. I was first expecting this Repo to be for OEM-connected devices. Like TTGO and other that have RF connected(soldered) to them when you get them.
If some DIY connected things on a breadboard are done, then its always up to the users to configure the pins they connected the RF to.

What is the opinion of others on this topic?

jgromes commented:
@serialrf433 this is definitely not as clear-cut as other devices, however, I think this is justified because the Pico has a well-defined shield format. So far I know only of Waveshare radio shield for it. So granted, it might have been better to name this "Waveshare LoRa Raspberry Pi Pico shield" (and we might revisit that naming if conflicts with other Pico shields in the future arise), but the code remains the same.

The same however does not apply to any other breadboard-based design.
